ctapipe/image/extractor.py

```diff
@@ -1287,18 +1289,127 @@ def __call__(
             is_valid=is_valid,
         )
 
-
 @lru_cache
-def exponential_scale_from_pulse_shape(camera: CameraDescription):
-    pulse_shape = camera.readout.reference_pulse_shape
-    sample_width = camera.readout.reference_pulse_sample_width
+def deconvolution_parameters(camera: CameraDescription, upsampling: int, window_width: int, window_shift: int):
+    assert(upsampling > 0)
```

Comment:
Please don't use asserts for checks on input parameters, raise the appropriate exceptions with a nice error message, e.g. in this case, use:

```
if upsampling < 0:
    raise ValueError(f"upsampling must be > 0, got {upsampling}")
```

Reply:
Ok.

A PositiveInteger trait would be even nicer for such parameters - does that exist?

Reply:
It's not documented, but the `traitlets.Int` actually takes a `min` and `max` option:

https://github.com/ipython/traitlets/blob/3d35d531f0386585f47b8af6b3e15343e4bdd48a/traitlets/traitlets.py#L2242-L2251

Reply:
Thanks; done.